ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -400,7 +401,9 @@ forkBlockProduction IS{..} =
                         -> (Point blk, BlockNo)
     prevPointAndBlockNo slot c = case c of
         Empty _   -> (genesisPoint, genesisBlockNo)
-        c' :> hdr -> case blockSlot hdr `compare` slot of
+        c' :> hdr
+          | Just{} <- nodeIsEBB hdr -> (headerPoint hdr, blockNo hdr)
```

Comment:
This can be a bit more explicit: check that the `blockSlot` is not `> slot` + add a comment saying that we allow forging a block in the same slot as an EBB.

Reply:
Fixed, thanks.